src/test/java/org/codeforamerica/messaging/models/TemplateTest.java

```diff
@@ -0,0 +1,24 @@
+package org.codeforamerica.messaging.models;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+public class TemplateTest {
+
+    @Test
+    void whenInputHasValidTemplateParamsThenReturnFilledTemplate() throws RuntimeException {
+        Map<String, Object> templateParams = Map.of("recipient_name", "Jarvis", "sender_name", "Papworth");
+        Template template = Template.builder()
+                .subject("Hi {{recipient_name}}! First SMS ever!")
+                .body("Merry Christmas {{recipient_name}}! From {{sender_name}}")
+                .build();
+
+        Assertions.assertEquals("Hi Jarvis! First SMS ever!",
+                template.buildSubjectFromTemplate(templateParams));
+        Assertions.assertEquals("Merry Christmas Jarvis! From Papworth",
+                template.buildBodyFromTemplate(templateParams));
+    }
+
```

Comment:
Test for invalid template params?

Reply:
One downside of handlebars seems to be that by default it does not throw an exception if params are missing, it just silently uses a blank string. I didn't want to write a test that asserts this behavior, since I don't think we want that. This is more material for the validation story, which I have created and added details here: https://www.pivotaltracker.com/story/show/185139322